library/ecp.c

```diff
@@ -1784,14 +1788,18 @@ int mbedtls_ecp_check_privkey( const mbedtls_ecp_group *grp, const mbedtls_mpi *
 #if defined(ECP_MONTGOMERY)
     if( ecp_get_type( grp ) == ECP_TYPE_MONTGOMERY )
     {
-        /* see [Curve25519] page 5 */
+        /* see [draft-irtf-cfrg-curves] page 7 */
```

Comment:
I think it's more readable and scalable if we have an `if` block per supported curve here, and signal an internal error if the curve ID is not covered. This way we make sure we'll always be looking at this code-path in case we're adding another Montgomery curve at some point.